src/core/evaluator.js

```diff
@@ -1306,6 +1306,47 @@ class PartialEvaluator {
     throw new FormatError(`Unknown PatternName: ${patternName}`);
   }
 
+  _parseVisibilityExpression(array, nestingCounter, currentResult) {
+    const MAX_NESTING = 10;
```

Comment:
I think it'd be *generally* safer to the do nesting check here, since that guarantees that it cannot be missed if any future changes are made to this method; hence something like this (which also removes the need to manually increment the counter in the call below):
```javascript
    if (++nestingCounter > MAX_NESTING) {
      warn("Visibility expression is too deeply nested");
      return;
    }
```

Reply:
That's a good idea